tests/load/lancedb/test_schema_alignment.py

```diff
@@ -0,0 +1,281 @@
+        assert_load_info(info)
+        # get data from destination
+        with pipeline.destination_client() as client:
+            table_name = client.make_qualified_table_name("all_types_table")  # type: ignore[attr-defined]
+            tbl = client.db_client.open_table(table_name)  # type: ignore[attr-defined]
+            actual_columns = set(tbl.schema.names)
+            if data_type == "json":
+                data_type = "json__a"
+            assert data_type in actual_columns, (
+                f"Expected {data_type} column to be present in destination table. Actual columns:"
+                f" {actual_columns}"
+            )
+            print("passed for data type", data_type)
+            # todo? check the actual datatype?
+
+
+@pytest.mark.parametrize("object_format", ["object", "pandas", "arrow-table"])
+def test_new_column_in_second_load(object_format: TestDataItemFormat) -> None:
+    """Test that new columns in source are added to the target."""
+    pipeline = dlt.pipeline(
+        pipeline_name="test_new_column_in_second_load",
+        destination="lancedb",
+        dataset_name=f"test_new_column_in_second_load_{uniq_id()}",
+        dev_mode=True,
+    )
+
+    @dlt.resource(
+        table_name="all_types_table",
+        primary_key="string",
+    )
+    def identity_resource(data: pa.Table) -> Generator[pa.Table, None, None]:
+        yield data
+
+    arrow_table: pa.Table = None
+    arrow_table, _, _ = arrow_table_all_data_types(
+        object_format=object_format, num_rows=1, include_decimal=False, include_json=False
+    )
+
+    info = pipeline.run(identity_resource(arrow_table))
+    assert_load_info(info)
+
+    # Add an extra column to the source
+    next_arrow_table, _, _ = arrow_table_all_data_types(
+        object_format=object_format, num_rows=1, include_decimal=True, include_json=False
+    )
+    info = pipeline.run(identity_resource(next_arrow_table))
+    assert_load_info(info)
+
+    schema_in_pipeline = pipeline.default_schema
+    assert "decimal" in schema_in_pipeline.tables["all_types_table"]["columns"]
+
+    # Verify that the extra column is in the actual destination table
+    with pipeline.destination_client() as client:
+        table_name = client.make_qualified_table_name("all_types_table")
+        tbl = client.db_client.open_table(table_name)
+
+        # Get the actual table schema from the destination
+        actual_columns = set(tbl.schema.names)
+
+        # Check that the decimal column was added
+        assert "decimal" in actual_columns, (
+            "Expected 'decimal' column to be present in destination table. Actual columns:"
+            f" {actual_columns}"
+        )
+
+
+# @pytest.mark.skip(reason="not completely understood what should happen")
+# this fails not sure why
+@pytest.mark.parametrize("object_format", ["object", "pandas", "arrow-table"])
+def test_missing_column_in_second_load(object_format: TestDataItemFormat) -> None:
```

Comment:
error
2025-07-03 12:26:04,556|[WARNING]|203593|130097760614208|dlt|load.py|complete_jobs:430|Job for all_types_table.41059b8a6b.reference retried in load 1751538360.4558077 with message lance error: LanceError(Schema): Schema error: No field named target__dlt_load_id. Valid fields are "?table?".string, "?table?".float, "?table?".int, "?table?".datetime, "?table?".bool, "?table?".pre_normalized_column, "?table?".time, "?table?".binary, "?table?".decimal, "?table?".date., /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/lance-0.30.0/src/dataset/write/merge_insert.rs:651:21 message:

Reply:
it happens because `_dlt_load_id` is not added to arrow table that is yielded. 
https://dlthub.com/docs/dlt-ecosystem/verified-sources/arrow-pandas#add-_dlt_load_id-and-_dlt_id-to-your-tables
this is required by "remove orphan" job

you need to enable this behavior. what we need to do here:
1. document this in docs
2. check this is `verify_schema` (for root tables, if remove orphan tables is enabled and `_dlt_load_id` is not present - raise an exception with an explanation)
3. parametrize this test so it removes orphans or not

Reply:
status: i have added code that emits a warning if there is a job with merge-key or remove-orphans (it would be better to check write_disposition but somehow I couldnt access it easily and also the test didnt work, so maybe I was wrong conceptually

test is failing for object and df:

![image](https://github.com/user-attachments/assets/aa83e240-f190-49fd-b8cf-e480e869e9a6)